### ISSUE TITLE
src/project/amplify/.config/local-env-info.json' error raised when deploy.

### ISSUE THREAD

**Issue by hyungjunk** [aws-amplify/amplify-cli#5931]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
`Error: File at path: '/codebuild/output/src663057245/src/REPO/amplify/.config/local-env-info.json' does not exist`
 error is raised when project is deployed by `amplifyPush --simple` command.

**Amplify CLI Version**
- 4.34.0

**To Reproduce**
Project is connected to github.
1. merge to dev branch from working branch
2. Triggers build script
```
backend:
  phases:
    build:
      commands:
        - chmod +x ./config/backend.sh
        - ./config/backend.sh
frontend:
...
```
3. build script is 
```
if [ "${AWS_BRANCH}" = "master" ] || [ "${AWS_BRANCH}" = "dev" ]
then
  echo "BUILD ENV:  ${AWS_BRANCH}"
  npm install -g @aws-amplify/cli
  amplifyPush --simple
else
  echo "BUILD ENV: ${AWS_BRANCH}"
  echo "Not pushing backend"
fi
```

**Expected behavior**
Deployed successfully

**Desktop (please complete the following information):**
WSL

**Additional context**

- Using custom build image `mrytkm/amplify-amazonlinux2`
- Log output
```
2020-11-23T01:44:12.252Z [INFO]: Git SSH Key acquired
2020-11-23T01:44:12.325Z [INFO]: # Cloning repository: git@github.com:TEAM/REPO.git
2020-11-23T01:44:12.338Z [INFO]: Agent pid 129
2020-11-23T01:44:12.344Z [INFO]: Identity added: /root/.ssh/git_rsa (/root/.ssh/git_rsa)
2020-11-23T01:44:12.416Z [INFO]: Cloning into 'MYPROJECT'...
2020-11-23T01:44:13.090Z [INFO]: Warning: Permanently added the RSA host key for IP address 'MYIP' to the list of known hosts.
2020-11-23T01:44:16.200Z [INFO]: # Switching to commit: 9bf07fd88ec507104956a3ac7d91e53d681aabef
2020-11-23T01:44:16.331Z [INFO]: Agent pid 142
2020-11-23T01:44:16.335Z [INFO]: Identity added: /root/.ssh/git_rsa (/root/.ssh/git_rsa)
2020-11-23T01:44:16.352Z [INFO]: Note: switching to '9bf07fd88ec507104956a3ac7d91e53d681aabef'.
                                 You are in 'detached HEAD' state. You can look around, make experimental
                                 changes and commit them, and you can discard any commits you make in this
                                 state without impacting any branches by switching back to a branch.
                                 If you want to create a new branch to retain commits you create, you may
                                 do so (now or later) by using -c with the switch command. Example:
                                 git switch -c <new-branch-name>
                                 Or undo this operation with:
                                 git switch -
                                 Turn off this advice by setting config variable advice.detachedHead to false
                                 HEAD is now at 9bf07fd Merge pull request #80 from REPO/BRANCH
2020-11-23T01:44:16.359Z [INFO]: Successfully cleaned up Git credentials
2020-11-23T01:44:16.359Z [INFO]: # Checking for Git submodules at: /codebuild/output/src663057245/src/REPO/.gitmodules
2020-11-23T01:44:16.389Z [INFO]: # Retrieving cache...
2020-11-23T01:45:10.883Z [INFO]: # Extracting cache...
2020-11-23T01:46:22.466Z [INFO]: # Extraction completed
2020-11-23T01:46:22.731Z [INFO]: # Retrieving environment cache...
2020-11-23T01:46:22.820Z [INFO]: # Retrieved environment cache
2020-11-23T01:46:22.828Z [INFO]: # Ensuring NPM package '@aws-amplify/cli' is version: 'latest'
2020-11-23T01:46:26.030Z [INFO]: # Patching NPM package '@aws-amplify/cli' from 1.12.0 to 4.34.0...
2020-11-23T01:47:20.904Z [INFO]: # Done patching NPM package '@aws-amplify/cli'
2020-11-23T01:47:25.225Z [INFO]: ## Starting Backend Build
                                 # Starting phase: build
                                 # Executing command: chmod +x ./config/backend.sh
2020-11-23T01:47:25.229Z [INFO]: # Executing command: ./config/backend.sh
2020-11-23T01:47:25.231Z [INFO]: AWS_BRANCH name: dev
2020-11-23T01:47:25.231Z [INFO]: BUILD ENV:  dev
2020-11-23T01:47:25.235Z [INFO]: # Getting Amplify CLI Cloud-Formation stack info from environment cache
2020-11-23T01:47:25.248Z [INFO]: # Start initializing Amplify environment: dev
2020-11-23T01:47:25.248Z [INFO]: {STACK INFO...} 
2020-11-23T01:47:26.347Z [INFO]: File at path: '/codebuild/output/src663057245/src/REPO/amplify/.config/local-env-info.json' does not exist
2020-11-23T01:47:26.348Z [INFO]: Error: File at path: '/codebuild/output/src663057245/src/REPO/amplify/.config/local-env-info.json' does not exist
                                 at Function.JSONUtilities.readJson (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:40:19)
                                 at StateManager.getData (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:166:56)
                                 at StateManager.getLocalEnvInfo (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:77:25)
                                 at teamProviderInfoGetAuthResourceNameHasSecrets (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/utils/team-provider-migrate.js:44:61)
                                 at Object.exports.migrateTeamProviderInfo (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/utils/team-provider-migrate.js:18:34)
                                 at Object.run (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/index.js:76:45)
                                 at process._tickCallback (internal/process/next_tick.js:68:7)
                                 at Function.Module.runMain (internal/modules/cjs/loader.js:832:11)
                                 at startup (internal/bootstrap/node.js:283:19)
                                 at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
2020-11-23T01:47:26.355Z [ERROR]: !!! Build failed
2020-11-23T01:47:26.355Z [ERROR]: !!! Non-Zero Exit Code detected
2020-11-23T01:47:26.355Z [INFO]: # Starting environment caching...
2020-11-23T01:47:26.355Z [INFO]: # Uploading environment cache artifact...
2020-11-23T01:47:26.459Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by chuongvuoop2021** [comment]
I have the same issue, anyone finds a workaround yet?

**Comment by hyungjunk** [comment]
I just had to downgrade amplify version. You can set in Uild settings menu.

**Comment by chuongvuoop2021** [comment]
> I just had to downgrade amplify version. You can set in Build settings menu.

which version do you use?

**Comment by hyungjunk** [comment]
It is 4.29.x

Can't rememeber precisely but pretty sure it wont be a big matter.

I recommend checking out the last successful build log of your project and use the same version. Thats what I did.

**Comment by chuongvuoop2021** [comment]
my last successful build was with 4.32.1, same as my fail one, I downgraded it back to 4.29.0, and still wouldn't work, found a workaround by not ignoring `amplify/.config/project-config.json` though, thanks

**Comment by ammarkarachi** [comment]
@hyungjunk @chuongvuoop2021 Hey all I could reproduce we are actively working on a fix now

**Comment by jjborie** [comment]
I have the same issue, the last version working for us is : 4.32.1

**Comment by javieramirez90** [comment]
Same here! Since yesterday morning. I tried by not ignoring amplify/.config/project-config.json but when I push to my production branch it gets stucked

**Comment by Roya1jr** [comment]
I had the same issue on 4.34.0 what fixed it for me was changing the Build image settings to use the latest Amplify cli because the default version seems to be older than the latest, previously I was updating the amplify cli as part of my build script but that wasn't working anymore. Also 4.34.0 has changed were the some information is located, I think you have to run any amplify command to make this change to your project I ran `amplify --version` and the dialog to update the files started. Also setup a [service role](https://docs.aws.amazon.com/amplify/latest/userguide/how-to-service-role-amplify-console.html)  if you haven's already

**Comment by aveeday** [comment]
> I have the same issue, the last version working for us is : 4.32.1

Worked for me, thanks.

I just changed config in amplify console => build settings -> Build image settings
<img width="1246" alt="Screenshot 2020-11-25 at 15 52 22" src="https://user-images.githubusercontent.com/1748693/100236576-87527680-2f36-11eb-971f-5e47282e4613.png">

Dear amplify developers, please check for deploying code created with old version of amplify before release new version. For the last one year working on a project that use amplify I had experienced a similar issue a few times and such a problem is unacceptable for production use of amplify.

**Comment by jhockett** [comment]
A fix has been released in Amplify CLI version 4.36.1. If you still see this issue, please comment below and we will re-open.

**Comment by drclohite** [comment]
same problem as before. now a few weeks since I have been able to update. Thank you. 

[log (85).txt](https://github.com/aws-amplify/amplify-cli/files/5610896/log.85.txt)

**Comment by sky4git** [comment]
### Reopen this issue

I have updated the amplify cli version to 4.36.2. This issue is still happening. What ever the fix was, clearly its not working.

**Comment by nickamckenna** [comment]
I also still have this problem on 4.36.2.

**Comment by ammarkarachi** [comment]
@drclohite 
What's the error you see?

**Comment by ammarkarachi** [comment]
https://github.com/aws-amplify/amplify-cli/issues/5974 @drclohite your issue is somewhat identical to the issue mentioned here +1 or follow up the thread there

**Comment by sky4git** [comment]
### Reopen this issue again

The issue is not identical. 
#5974 happens when using **amplify env checkout dev**  command.

However, this happens with **amplify push --yes** in amplify.yml.  This is a more urgent issue than the other one, as currently it stops the production deployment pipeline.  
We have this simple command to push our changes to production using our production deployment script in amplify.yml.  We have noticed this issue with the latest amplify cli update. Currently, we cant deploy to production as it always fails. 

Either give some work around or fix this issue.  Please do not close this issue without proper solution. 

Issue is exactly the same as describe in very first comment.

**Comment by ammarkarachi** [comment]
@sky4git 
Can you share the log output?

**Comment by sky4git** [comment]
`2020-11-29T06:25:12.623Z [INFO]: # Executing command: amplify --version
2020-11-29T06:25:13.646Z [INFO]: Initializing new Amplify CLI version...
2020-11-29T06:25:13.939Z [INFO]: Done initializing new version.
2020-11-29T06:25:13.939Z [INFO]: Scanning for plugins...
2020-11-29T06:25:13.973Z [INFO]: Plugin scan successful
2020-11-29T06:25:14.006Z [INFO]: File at path: '/codebuild/output/src654173097/src/MYPROJECTNAME/amplify/.config/local-env-info.json' does not exist
2020-11-29T06:25:14.006Z [INFO]: Error: File at path: '/codebuild/output/src654173097/src/MYPROJECTNAME/amplify/.config/local-env-info.json' does not exist
                                 at Function.JSONUtilities.readJson (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:40:19)
                                 at StateManager.getData (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:166:56)
                                 at StateManager.getLocalEnvInfo (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:77:25)
                                 at teamProviderInfoGetAuthResourceNameHasSecrets (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/utils/team-provider-migrate.js:39:61)
                                 at Object.exports.migrateTeamProviderInfo (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/utils/team-provider-migrate.js:18:34)
                                 at Object.run (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/lib/index.js:78:45)
2020-11-29T06:25:14.013Z [ERROR]: !!! Build failed
2020-11-29T06:25:14.013Z [ERROR]: !!! Non-Zero Exit Code detected`

**Comment by jhockett** [comment]
@sky4git Can you try changing your amplify CLI version in the console to 4.32.1? https://github.com/aws-amplify/amplify-cli/issues/5931#issuecomment-733724542

**Comment by sky4git** [comment]
I seen that workaround in previous comment. But its not a proper fix. 
Why would we need to rollback to previous CLI version. 
It creates inconsistencies in local and production environments. 
There should be a proper fix for this. 

If we have to do that, that should be the last option.

**Comment by ammarkarachi** [comment]
The reason why this issue is closed is because we have skipped checks for env on `amplify init, pull and env` in 4.36.2. and the [amplifyPush script](https://github.com/aws-amplify/amplify-console/blob/master/scripts/amplifyPush.sh) does basically that

**Comment by ammarkarachi** [comment]
This is the line that was changed, by PR https://github.com/aws-amplify/amplify-cli/pull/5987/files
https://github.com/aws-amplify/amplify-cli/blob/7d48378bbc01d0c104670de2ee40abb532f203cc/packages/amplify-cli/src/utils/team-provider-migrate.ts#L36

**Comment by jhockett** [comment]
> I seen that workaround in previous comment. But its not a proper fix.
> Why would we need to rollback to previous CLI version.
> It creates inconsistencies in local and production environments.
> There should be a proper fix for this.
> 
> If we have to do that, that should be the last option.

Sorry for being unclear. I'm trying to provide a temporary workaround so that you can get unblocked. I didn't mean to imply it was a replacement for a fix.

**Comment by sky4git** [comment]
@ammarkarachi 
Yes, I understand the change. 
But what is the fix?
How can I make my production pipeline working again? What do I need to change in my amplify.yml script?

Remember, this issue is happening without we doing anything wrong and just updating the CLI version caused this issue.

@jhockett My local cli version is already upgraded to 4.36.2. no issue when using amplify push --yes. Its just production pipeline is broken. I am afraid to rollback to previous version now. It can cause unintended effects on production.

**Comment by ammarkarachi** [comment]
Can you confirm the version on your production pipeline?

**Comment by sky4git** [comment]
4.36.2

**Comment by ammarkarachi** [comment]
@sky4git What's your build script like?

**Comment by sky4git** [comment]
```yaml
version: 1
env:
   AMPLIFY_AMAZON_CLIENT_ID: "XXX"
  AMPLIFY_AMAZON_CLIENT_SECRET: "XXX"
  AMPLIFY_FACEBOOK_CLIENT_ID: "XXX"
  AMPLIFY_FACEBOOK_CLIENT_SECRET: "XXX"
  AMPLIFY_GOOGLE_CLIENT_ID: "XXX"
  AMPLIFY_GOOGLE_CLIENT_SECRET: "XXX"
  MODE: "production"
#ENV: "master"
backend:
  phases:
    preBuild:
      commands:
        - yum install golang -y
        - npm install -g @aws-amplify/cli
    build:
      commands:
        - ls -la
        - amplify --version
        - go env
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - npm ci
    # IMPORTANT - Please verify your build commands
    build:
      commands:
        - npm run export
  artifacts:
    # IMPORTANT - Please verify your build output directory
    baseDirectory: out
    files:
      - "**/*"
  cache:
    paths:
      - node_modules/**/*

```

**Comment by nickamckenna** [comment]
Downgrading to 4.32.1 solved my problem!

**Comment by drclohite** [comment]
[log (95).txt](https://github.com/aws-amplify/amplify-cli/files/5619539/log.95.txt)
Attached is the latest error file, same as past error files. I cannot get past 4.32 as a console setting, although on my pc I have the latest version running. Operationally I see odd database results. When I run dynamodb searches on console I get a good result. When I run an identical search in code I get an off result, usually half the data i.e. 2 records instead of the 4 expected. data size is no size at all, this is not a size issue. If I take away search criteria I get a full 100 records. However, the system has more than 100 records so I cannot simply sift through the 100 as a workaround.  my dev machine is also rolled back to 4.32 although it works set to latest. The problem is only when I update my github files and Amplify Build Image is set to latest.

**Comment by ammarkarachi** [comment]
@drclohite Do you know what your role policies are?

**Comment by drclohite** [comment]
I have administrator access which includes Amplify and Amplify Admin

**Comment by drclohite** [comment]
any progress? Still building to 4.32, although no hold up

**Comment by yutaoyan** [comment]
Hey folks, please fix this problem! As of now, I am afraid of developing any new features for my app on Amplify. If this problem still exists for a couple of months, we might as well just call 4.32.1 is the final version of Amplify. Any new features release on Amplify will not be used for any production environment anyway.

**Comment by aspearman** [comment]
I created an Amplify app using CLI v4.39.0 on one computer, but git pulling my iOS codebase and trying to run `amplify configure` with the same CLI version on **different** computer results in the error:

`File at path: '/Users/.../amplify/.config/local-env-info.json' does not exist`

That file is ignored in the default `.gitignore` given by Amplify, so this really does seem like a bug. 

From what I understand, I shouldn't use a CLI version lower than the one I created the project with. Actually, I get a different error when I try downgrading to 4.32.1 and running `amplify configure`:

`File project: data.graphqltransformer should NOT have additional properties: 'enableiterativegsiupdates'`

This is extremely frustrating!

**Comment by ammarkarachi** [comment]
Hi all, we are working on a release to fix this problem as soon as possible, I apologize for the issues we are working on it to get the issues addressed as soon as possible.

**Comment by sky4git** [comment]
This original bug seems to be fixed now with cli version **@aws-amplify/cli@4.40.0.**

However, the **old bug is resurfaced now again**.
`Error: auth headless is missing the following inputParams googleClientId, facebookAppId`

**I have all the necessary environment variables in my environment variables settings:**
AMPLIFY_FACEBOOK_CLIENT_ID
AMPLIFY_FACEBOOK_CLIENT_SECRET
AMPLIFY_GOOGLE_CLIENT_ID
AMPLIFY_GOOGLE_CLIENT_SECRET
You can see my amplify.yml script again here: https://github.com/aws-amplify/amplify-cli/issues/5931#issuecomment-735527370

Here is the whole log for your information:
`2020-12-19T06:08:55.477Z [INFO]: [31mFailed to pull the backend.[39m
2020-12-19T06:08:55.480Z [INFO]: [31mauth headless is missing the following inputParams googleClientId, facebookAppId[39m
2020-12-19T06:08:55.485Z [INFO]: [0mError: auth headless is missing the following inputParams googleClientId, facebookAppId[0m
                                 [0m    at updateConfigOnEnvInit (/root/.nvm/versions/node/v12.19.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/provider-utils/awscloudformation/index.js:135:15)[0m
                                 [0m    at /root/.nvm/versions/node/v12.19.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/index.js:300:28[0m
                                 [0m    at /root/.nvm/versions/node/v12.19.0/lib/node_modules/@aws-amplify/cli/node_modules/promise-sequential/index.js:16:18[0m
                                 [0m    at runMicrotasks (<anonymous>)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[0m
2020-12-19T06:08:55.506Z [ERROR]: !!! Build failed
2020-12-19T06:08:55.507Z [ERROR]: !!! Non-Zero Exit Code detected`

![image](https://user-images.githubusercontent.com/8438024/102682378-2c522d80-421d-11eb-860d-37e395e15fa6.png)

This is also breaking production **amplifyPush --simple** and not an issue with local environment **amplify push**. 
**If you want I can create a new ticket but please fix this permanently**

**Comment by drclohite** [comment]
Yup, 4.4 did not work for me either. I had to drop from 4.32 to 4.30 to get my side working again. Phew!

**Comment by arturlr** [comment]
The only way I got it working with the 4.41.1 version was creating a local-env-info.json with {} in its content.

**Comment by danielmartincraig** [comment]
I'm seeing this exact issue w version 4.41.1

**Comment by smithi1** [comment]
I am also seeing this with 4.41.1 - fixing the Amplify CLI version to 4.40.0 in the build image settings has cured it.

**Comment by cornwe19** [comment]
I've also run into this with 4.41.1. Thanks for the suggestion, @smithi1

**Comment by yutaoyan** [comment]
I tested on 4.41.2 and it WORKS without any workaround, but someone else please confirms this.

**Comment by aspearman** [comment]
@yutaoyan @ammarkarachi `4.41.2` worked when pulling an environment. Success!

**Comment by yutaoyan** [comment]
@aspearman But the better question now is what fixed this?! I double-checked all recent PRs, nothing hinted me is targeted at this issue.

**Comment by hbj** [comment]
I have the same issue with 4.41.2 when running `amplify init` on a project created by a colleague that I cloned from a git repo. Using 4.40.0 as suggested by @smithi1 fixed the issue.

**Comment by Stvad** [comment]
It's interesting to note that the file exists locally, but it's in default `.gitignore`. creating dummy one during build helped.  
and then I hit #5791 straight away

**Comment by codrin-lanterne** [comment]
I've also ran into this with 4.41.2 used locally and in the build image settings but chaning the build image settings version to 4.40.0 fixed it.

**Comment by drclohite** [comment]
fault still exists, not sure why this is closed.

# Importing Amplify environment: dev (amplify env import)
2021-01-30T17:30:21.665Z [INFO]: [32mSuccessfully added environment from your project[39m
2021-01-30T17:30:22.104Z [INFO]: # Initializing existing Amplify environment: dev (amplify init)
2021-01-30T17:30:24.516Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-01-30T17:30:30.765Z [WARNING]: - Initializing your environment: dev
2021-01-30T17:30:30.831Z [INFO]: [31mCould not initialize 'dev': Access Denied[39m
2021-01-30T17:30:32.738Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-01-30T17:30:37.790Z [WARNING]: - Initializing your environment: dev
2021-01-30T17:30:37.846Z [INFO]: [31mCould not initialize 'dev': Access Denied[39m
2021-01-30T17:30:37.864Z [ERROR]: !!! Build failed
2021-01-30T17:30:37.864Z [ERROR]: !!! Non-Zero Exit Code detected
2021-01-30T17:30:37.865Z [INFO]: # Starting environment caching...
2021-01-30T17:30:37.866Z [INFO]: # Uploading environment cache artifact...
2021-01-30T17:30:37.927Z [INFO]: # Environment caching completed

**Comment by drclohite** [comment]
Ok, it works for 4.3. What am I looking at in terms of IAM?

**Comment by constantineh** [comment]
Getting the following fault with amplify 4.45.2. Localy works fine with same version.
I have tried various versions but nothing fixes the issue, any ideas?

`                                 # Starting phase: build
                                 # Executing command: amplifyPush --simple
2021-03-16T21:46:47.931Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-03-16T21:46:55.855Z [INFO]: [0mError: AWS access credentials can not be found.[0m
                                 [0m    at newUserCheck (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:628:13)[0m
                                 [0m    at Object.init (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/configuration-manager.ts:70:11)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[0m
                                 [0m    at Object.run (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/initializer.js:19:3)[0m
2021-03-16T21:46:55.855Z [HELP]: You do not have a role attached to your app
2021-03-16T21:46:55.872Z [ERROR]: !!! Build failed
2021-03-16T21:46:55.873Z [ERROR]: !!! Non-Zero Exit Code detected
2021-03-16T21:46:55.873Z [INFO]: # Starting environment caching...
2021-03-16T21:46:55.873Z [INFO]: # Environment caching completed
Terminating logging...`

**Comment by samuelmburu** [comment]
This issue cropped up for me, and I'm using `@aws-amplify/cli@4.45.2` -- things were working a couple of days ago.

Today, I opened up my terminal and ran `amplify status` and got the original poster's error.

Building locally, I tried to "fix" things by putting all of my uncommitted changes onto another branch. I then went back to my previous branch and performed a `git pull` but could not run `amplify status` successfully.

I then decided to clone my repo again in another directory, but I still have the same result. 

----------

Then when I run the amplify I had in my project `"aws-amplify": "^3.0.20",` via  (npx amplify status), I get this:
```shell
Scanning for plugins...
Plugin scan successful
| Category | Resource name                         | Operation | Provider plugin   |
| -------- | ------------------------------------- | --------- | ----------------- |
| Function | breeeerrr823749729020300CustomMessage | Create    | awscloudformation |
| Auth     | browuud8xtensionbe8b2ad1              | Create    | awscloudformation |
| Api      | browuud8xtefoper                      | Create    | awscloudformation |

```

This is wild because all of these services already exist. After all, I have an application already running using this configuration.

I'll try running another version of the cli

**Comment by samuelmburu** [comment]
Luckily my other machine had v3.9.1 that still works
